CMakeLists.txt

```diff
@@ -335,8 +335,10 @@ if("${ENABLE_ALL_WARNINGS}")
          -Wno-strict-aliasing \
          -Wno-type-limits \
          -Wno-stringop-overflow \
-         -Wno-stringop-overread \
-         -Wno-return-type")
+         -Wno-stringop-overread")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.2)
+      set(KNOWN_COMPILER_SPECIFIC_WARNINGS "-Wno-return-type")
```

Comment:
Not sure I still fully comprehend why this is only turned on for compilers > 12.2 (this doesnt even check what kind of compiler). The change referenced in the discription was a folly failure, If we have warnings that are caught by this in Velox , we should fix them.

Reply:
I did globally disable but this PR https://github.com/facebookincubator/velox/pull/4618 specifies Fedora with GCC 12.2 and I don't have this environment to test what no return compilation failures are thrown. The PR has fixed whatever errors are thrown by the present Velox PR pipelines.

Reply:
Removed this bit.